x-pack/plugins/security_solution/public/detections/pages/alert_details/tabs/summary/case_panel.test.tsx

```diff
@@ -0,0 +1,185 @@
+        <TestProviders>
+          <CasePanel
+            eventId={mockAlertNestedDetailsTimelineResponse._id}
+            dataAsNestedObject={mockAlertNestedDetailsTimelineResponse}
+            detailsData={mockAlertDetailsTimelineResponse}
+          />
+        </TestProviders>
+      );
+
+      expect(screen.getByText(ERROR_LOADING_CASES)).toBeVisible();
+    });
+
+    describe('Partial permissions', () => {
+      it('should only render the add to new case button', () => {
+        (useGetRelatedCasesByEvent as jest.Mock).mockReturnValue({
+          loading: false,
+          relatedCases: [],
+        });
+        (useGetUserCasesPermissions as jest.Mock).mockReturnValue({
+          create: true,
+          update: false,
+        });
+        render(
+          <TestProviders>
+            <CasePanel
+              eventId={mockAlertNestedDetailsTimelineResponse._id}
+              dataAsNestedObject={mockAlertNestedDetailsTimelineResponse}
+              detailsData={mockAlertDetailsTimelineResponse}
+            />
+          </TestProviders>
+        );
+
+        expect(screen.getByText(ADD_TO_NEW_CASE_BUTTON)).toBeVisible();
+        expect(screen.queryByText(ADD_TO_EXISTING_CASE_BUTTON)).toBe(null);
+      });
+
+      it('should only render the add to existing case button', () => {
+        (useGetRelatedCasesByEvent as jest.Mock).mockReturnValue({
+          loading: false,
+          relatedCases: [],
+        });
+        (useGetUserCasesPermissions as jest.Mock).mockReturnValue({
+          create: false,
+          update: true,
+        });
+        render(
+          <TestProviders>
+            <CasePanel
+              eventId={mockAlertNestedDetailsTimelineResponse._id}
+              dataAsNestedObject={mockAlertNestedDetailsTimelineResponse}
+              detailsData={mockAlertDetailsTimelineResponse}
+            />
+          </TestProviders>
+        );
+
+        expect(screen.getByText(ADD_TO_EXISTING_CASE_BUTTON)).toBeVisible();
+        expect(screen.queryByText(ADD_TO_NEW_CASE_BUTTON)).toBe(null);
+      });
+
+      it('should render both add to new case and add to existing case buttons', () => {
+        (useGetRelatedCasesByEvent as jest.Mock).mockReturnValue({
+          loading: false,
+          relatedCases: [],
+        });
+        (useGetUserCasesPermissions as jest.Mock).mockReturnValue({
+          create: true,
+          update: true,
+        });
+        render(
+          <TestProviders>
+            <CasePanel
+              eventId={mockAlertNestedDetailsTimelineResponse._id}
+              dataAsNestedObject={mockAlertNestedDetailsTimelineResponse}
+              detailsData={mockAlertDetailsTimelineResponse}
+            />
+          </TestProviders>
+        );
+
+        expect(screen.queryByText(ADD_TO_NEW_CASE_BUTTON)).toBeVisible();
+        expect(screen.queryByText(ADD_TO_EXISTING_CASE_BUTTON)).toBeVisible();
+      });
+    });
+  });
+  describe('has related cases', () => {
```

Comment:
The cases team is working to provide us the case description, status, and other fields, so these tests will be expanded once we have that information